docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerSSHLauncher.java

```diff
@@ -150,7 +153,11 @@ protected InetSocketAddress getAddressForSSHD(String cloudId, InspectContainerRe
              * like Joyent's Triton. */
                 if (host == null || host.equals("0.0.0.0") || usesSingleHostAbstraction(ir)) {
                     // Try to connect to the container directly (without going through the host)
-                    host = networkSettings.getIpAddress();
+                    if (network != null && network.length() > 0) {
+                        host = networkSettings.getNetworks().get(network).getIpAddress();
```

Comment:
That wouldn't work either when the Jenkins master is running in a Docker container itself. Slaves ports are exposed through the Docker hostname and not on the virtual network IP. I tried your PR and didn't work for me. I actually needed to add a new "Docker hostname" configuration to specifically tell what the physical IP looks like. See https://github.com/jenkinsci/docker-plugin/pull/443